src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -360,19 +360,32 @@ public static bool IsEntirelyWithinCrefSyntax(this SyntaxTree syntaxTree, int po
 
         private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int position, char lastChar, CancellationToken cancellationToken)
         {
-            if (token.Kind() is not (
+            var kind = token.Kind();
+            if (kind is not (
                     SyntaxKind.StringLiteralToken or
                     SyntaxKind.CharacterLiteralToken or
                     SyntaxKind.SingleLineRawStringLiteralToken or
-                    SyntaxKind.MultiLineRawStringLiteralToken))
+                    SyntaxKind.MultiLineRawStringLiteralToken or
+                    SyntaxKind.Utf8StringLiteralToken or
+                    SyntaxKind.Utf8SingleLineRawStringLiteralToken or
+                    SyntaxKind.Utf8MultiLineRawStringLiteralToken))
             {
                 throw new ArgumentException(CSharpCompilerExtensionsResources.Expected_string_or_char_literal, nameof(token));
             }
 
+            // A UTF8 string literal must end with `"u8` therefore never incomplete.
```

Comment:
```suggestion
            // A UTF8 string literal must end with `"u8` and is thus never incomplete.
```

Reply:
I had to ask ChatGPT for the difference here 🤣

> Both of these statements convey similar information, but there are some differences in their wording and tone:
> 
> 1. "A UTF8 string literal must end with `"u8` therefore never incomplete."
> 
> This sentence uses the word "therefore" to establish a logical connection between the requirement for a UTF8 string literal to end with `"u8` and the assertion that it is never incomplete. The word "therefore" indicates that the completion of the string literal with `"u8` is a result of the requirement and consequently ensures that the literal is never incomplete. This statement is more direct and formal in its presentation.
> 
> 2. "A UTF8 string literal must end with `"u8` and is thus never incomplete."
> 
> In this sentence, the phrase "and is thus" is used to indicate a consequence or outcome of the requirement that the string literal must end with `"u8`. The word "thus" suggests a cause-and-effect relationship, implying that the requirement guarantees the completeness of the string literal. This statement has a slightly more explanatory or descriptive tone compared to the first one.
> 
> In terms of English usage, both sentences are grammatically correct and convey the same essential idea: the requirement for a UTF8 string literal to end with `"u8` ensures that the literal is never incomplete. The choice between them might depend on the context and the desired tone of communication.

Reply:
So `A UTF8 string literal must end with "u8 therefore never incomplete.` reads very wrong to me.  The lack of `and is` before `therefor` just sounds wrong afaict.

i'm fine with `and is therefore never incomplete` or `and is thus never incomplete` are both ok to me.